### PR TITLE
Add peb documentation

### PR DESCRIPTION
**Pull request by juliensnz**

### CODE REVIEW

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
+Create a custom product export builder filter
+=============================================
+
+When you create a new attribute type or if you want to change the rendering of a filter you need to create a custom filter. In this quick cookbook we we go throug all the step to acheive this task.
```
Comment (rybus) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: we will

Comment (rybus) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: all the steps

Comment (rybus) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: to achieve

Comment (rybus) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: through

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+Declare a filter provider for your custom filter:
++++++++++++++++++++++++++++++++++++++++++++++++++
+
+To create a custom filter, first we need to create a FilterProvider for our new filter type:
+
+.. code-block:: php
+    :linenos:
+    <?php
+
+    /*
```
Comment (rybus) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: Is this only available for EE?

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    parameters:
+        acme.custom.provider.filter.range.class: Acme\Bundle\CustomBundle\Enrich\Provider\Filter\RangeFilterProvider
+
+    services:
+        acme.custom.provider.filter.range:
+            class: %acme.custom.provider.filter.range.class%
+            tags:
+                - { name: pim_enrich.provider.filter, priority: 110 }
+
+Your filter provider is now registered, congrats!
```
Comment (rybus) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: I think "congrats" is a bit too much :p

File: cookbook/ui_customization/create_a_new_filter_type.rst (hunk 2 above)
Comment (willy-ahva) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: you should remove the licence

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+.. code-block:: yaml
+    :linenos:
+
+    # Acme/Bundle/CustomBundle/Resources/config/form_extensions.yml
+
+    extensions:
+        acme-range-filter:
+            module: acme/range-filter
+
+After a cache clear, you can set the min and max value of any number attribute to start to use this new custom filter!
```
Comment (rybus) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: after a cache clearing / after you cleaned the cache

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+    use Pim\Bundle\EnrichBundle\Provider\Filter\FilterProviderInterface;
+    use Pim\Component\Catalog\Model\AttributeInterface;
+
+    class RangeFilterProvider implements FilterProviderInterface
+    {
+        /**
+         * {@inheritdoc}
+         */
+        public function getFilters($attribute)
```
Comment (willy-ahva) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: I think you should explain what methods do.

**Comment by rybus** [comment]
Sometimes you use "Your stuff" and then "our stuff", you should remain consistent between those two. Otherwise, a few changes to make and then GTM :)

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        }
+    }
+
+Now we can register this filter provider in our service.yml file
+
+.. code-block:: yaml
+    :linenos:
+
+    parameters:
+        acme.custom.provider.filter.range.class: Acme\Bundle\CustomBundle\Enrich\Provider\Filter\RangeFilterProvider
```
Comment (willy-ahva) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: Don't use a parameter. It's easier for integrators and not necessary.

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 7]
```diff
@@ -0,0 +1,156 @@
+Create a custom product export builder filter
+=============================================
+
+When you create a new attribute type or if you want to change the rendering of a filter you need to create a custom filter. In this quick cookbook we we go throug all the step to acheive this task.
+
+Declare a filter provider for your custom filter:
```
Comment (rybus) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: you can remove the colon at the end.

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+                - { name: pim_enrich.provider.filter, priority: 110 }
+
+Your filter provider is now registered, congrats!
+
+Create the filter
++++++++++++++++++
+
+Now that we have a filter provider, we can create the filter itself:
+
+'use strict';
```
Comment (willy-ahva) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: no ..code-block:: ?

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+    'pim/filter/attribute/attribute',
+    'text!acme/template/product/filter/range',
+    'jquery.select2'
+], function (
+    _,
+    __,
+    AttributeFilter,
+    template
+) {
+    return AttributeFilter.extend({
```
Comment (willy-ahva) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: You should briefly explain your code. A block of code is not really interesting if it's not explained in a cookbook IMHO.

File: cookbook/ui_customization/create_a_new_filter_type.rst (hunk 3 above)
Comment (willy-ahva) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: Not agree, it's fun and playful :) I like it !

File: cookbook/ui_customization/create_a_new_filter_type.rst (hunk 2 above)
Comment (juliensnz) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: yeah, you are right, a little mistake :)

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+        __,
+        AttributeFilter,
+        template
+    ) {
+        return AttributeFilter.extend({
+            shortname: 'range',
+            template: _.template(template),
+
+            /*
+            We listen on the change event of the range field.
```
Comment (willy-ahva) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: miss a star here

**Comment by willy-ahva** [comment]
GTM :+1:

**Comment by rybus** [comment]
GTM

File: cookbook/ui_customization/create_a_new_filter_type.rst (hunk 10 above)
Comment (juliensnz) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: nope it's not a method comment :)

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 11]
```diff
@@ -0,0 +1,155 @@
+Create a custom product export builder filter
+=============================================
+
+When you create a new attribute type or if you want to change the rendering of a filter you need to create a custom filter. In this quick cookbook we will go through all the steps to achieve this task.
```
Comment (MarieMinasyan) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: filter,

File: cookbook/ui_customization/create_a_new_filter_type.rst (hunk 10 above)
Comment (MarieMinasyan) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: listen to

File: cookbook/ui_customization/create_a_new_filter_type.rst [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+
+And its template:
+
+.. code-block:: html
+    :linenos:
+
+    <!-- src/Acme/Bundle/CustomBundle/Resources/public/templates/product/filter/range.html -->
+    <input type="range" name="filter-value" value="<%= value %>" min="<%= attribute.number_min %>" max="<%= attribute.number_max %>" <%- editable ? '' : 'disabled' %>/>
+
+You can now register this module into our requirejs configuration:
```
Comment (MarieMinasyan) [inline cookbook/ui_customization/create_a_new_filter_type.rst]: in

**Pull request merged**